solutions/cses-1160.mdx

```diff
@@ -2,62 +2,30 @@
 id: cses-1160
 source: CSES
 title: Planets Queries II
-author: Michael Cao, Benjamin Qi
+author: Dustin Miao, Michael Cao, Benjamin Qi
 ---
 
-In this problem, we are given a directed graph and asked $q$ queries for the
-minimum distance between two vertices $u$ and $v$.
+In this problem, we are given a functional directed graph and asked $q$ queries for the minimum distance between two vertices $u$ and $v$, if it exists. 
 
-# Main Idea
+## Solution
 
-The graph is a functional graph, so break it into tree / cycle cases and solve
-each one independently.
+We will decompose the graph into several cases. Note that because each node has one outgoing edge, there must be exactly one cycle in every connected component. 
```

Comment:
```suggestion
Decompose the graph into several cases. Note that because each node has one outgoing edge, there must be exactly one cycle in every connected component. 
```